common/array.h

```diff
@@ -296,26 +310,32 @@ class Array {
 		return true;
 	}
 
+    /** Check if two arrays are different. */
 	bool operator!=(const Array<T> &other) const {
 		return !(*this == other);
 	}
 
+    /** Return a pointer to the underlying memory where the array begins. */
```

Comment:
Again, I would not mention `underlying memory` here.
I would suggest: `Return an iterator pointing to the first element in the array.`